### DISCUSSION TITLE
Is this available as a javascript file?

### DISCUSSION THREAD

**Discussion by btacordex** [js-fns/date-fns#2750] [Q&A]
Moment is recommending this library as a possible replacement, but I'm not really interested in setting up npm and a new project just to be able to do format('DDDD'). Is there a downloadable .js file somewhere?

**Comment by ifrguy** [comment]
I don't have an answer, but way back date-fns ( v1.35  and before) was delivered with an additional monolithic webpack bundle, this is what I use to this day since our db shell is incapable of anything else.  I've been trying to build a single pack with webpack but so far I haven't been able to crack that nut.

**Reply by fturmel** [reply]
@ifrguy It's counter to the whole tree-shaking approach of v2, but one way to go about this could be to bundle an entry point that imports all functions and exposes them with a global variable. You could then load the file with a script tag and reference date-fns from anywhere in your app. You will most likely lose all type hints / IDE support when using the library though, which is a major downside.

Here's an example with `esbuild`. Maybe you can reference it as a jumping-off point.

Create the following `package.json` file and install the dependencies:
```json
{
  "dependencies": {
    "date-fns": "^2.26.0",
    "esbuild": "^0.13.14"
  },
  "scripts": {
    "buildLib": "echo 'import * as tmp from \"date-fns\"; window.dateFns = tmp;' | esbuild --bundle --minify --charset=utf8 --outfile=date-fns.js",
    "buildLocales": "echo 'import * as tmp from \"date-fns/locale\"; window.dateFnsLocales = tmp;' | esbuild --bundle --minify --charset=utf8 --outfile=date-fns-locales.js"
  }
}
```

Run the scripts, they should generate the following files:
- `date-fns.js` (~20.5 kB gzipped) includes all functions and `en-US` default locale, accessible using the `dateFns` global variable
- `date-fns-locales.js` (~117 kB gzipped), includes all locales, accessible using the `dateFnsLocales` global variable (optional)

HTML demo:
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <title>date-fns v2 bundle demo</title>
  </head>
  <body>
    <script src="date-fns.js"></script>
    <script src="date-fns-locales.js"></script>
    <script>
      const res = dateFns.format(new Date(), "PPPPpppp", {
        locale: dateFnsLocales.frCA,
      });
      console.log(res); // vendredi 19 novembre 2021 à 21:58:47 GMT-05:00
    </script>
  </body>
</html>
```

**Reply by ifrguy** [reply]
I understand the appeal of the modular approach, I'm working in a database platform that does not implement js module semantics. The only thing I can do is load js blobs.  It's all rather maddening.    Thanks for the suggestion. I'll play it, maybe I'm missing something.

**Reply by ifrguy** [reply]
fturmel;

After many a false start I have a single blob. Between my lack of knowledge and the assumptions you made, but I did finally crack it.  One thing I didn't understand right off the mark was the assumption that I was working from and installed version of date-fns and not the source package from github.  Finally guessed that.  Second assumption I'm working withing a browser environment, hence the "window.dateFns" assignment.  I'm not, mongoDB has no idea about window object, it just understands pure flat js, no DOM, or anything else other than database objects.  Any way with all that I was able to tease out the basic ideas and arrive where I needed to be, that is a single js date functions object that loads.  The locales are not an issue for me, so the base date-fns blob will suffice nicely.  The process of getting there adds a bunch of work to my day.  It would be nice if the dev's provided a "how to build" readme and maybe a build target to make a date-fns blob. 

Here's the recipe I derived from your detailed hint:
1. install the latest date-fns library with npm
2.  install esbuild with npm
3. create a package.json file with the necessary esbuild script
4. run npm build script(s) to get the js blob(s)

Thanks for the guidance.

ifrguy
--
Always keep the greasy side down.
